### DISCUSSION TITLE
Large Difference between BldPitch and BlPitchC with Simulink Controller

### DISCUSSION THREAD

**Discussion by JanHeimsoth** [OpenFAST/openfast#1652] [Q&A]
Dear all,

I am using the FAST_SFunc from the v.3.1.0 release to build my own controller for a small WT model I made with openFAST. I use a PI controller to control the pitch of the blades, but whenever I Pitch a little too quickly the BldPitch1-3 overshoot dramatically and do not match the input the controller gave.  This results in extreme changes in rotational speed and thus again in BlPitchC1-3 amplifying the problem even further. How could this be explained and/or fixed? Optimally I would want all three, my control input, BlPitchC1-3 and BldPitch1-3 to be the same.

Thank you already for your help and Best Regards
Jan Heimsoth

![20230626_BldPitch1](https://github.com/OpenFAST/openfast/assets/101802956/b8ca198f-9b67-4079-b044-4a473c2370a5)

**Comment by jjonkman** [comment]
Dear @JanHeimsoth,

Sounds similar to questions asked and answered on the forum, e.g., see: https://forums.nrel.gov/t/different-pitch-rate-in-and-outside-fast-sfunction-in-simulink/2316.

Best regards,
